src/components/App.js

```diff
@@ -1,32 +1,33 @@
-import PropTypes from 'prop-types';
+import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
 import Calculate from '../Logic/calculate';
 
-export default function App(props) {
-  const {
-    total, next, operation, buttonName,
-  } = props;
-  const obj = { total, next, operation };
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
 
-  return (
-    <>
-      <div className="display-container"><Display /></div>
-      <div className="button-panel-container"><ButtonPanel onClick={Calculate(obj, buttonName)} /></div>
-    </>
-  );
-}
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
 
-App.propTypes = {
-  total: PropTypes.string,
-  next: PropTypes.string,
-  operation: PropTypes.string,
-  buttonName: PropTypes.string,
-};
+  handleClick(buttonName) {
```

Comment:
### Optional
- [ ] It would be better if we create an error message for the division by zero logic. Currently, this breaks the application

Reply:
Thanks for the suggestion. I've fixed that error and merging the PR now :100: